### ISSUE TITLE
L014 triggered by leading underscore

### ISSUE THREAD

**Issue by j-svensmark** [sqlfluff/sqlfluff#3780]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing a file with sqlfluff, I unexpectedly got a `Unquoted identifiers must be consistently capitalised.` error from L014. Here is an MWE
```
select
    _a,
    b
from foo
```

### Expected Behaviour

I expect running sqlfluff on the above code sample shouldn't produce any error.

### Observed Behaviour

When running `sqlfluff lint --dialect bigquery MWE.sql` I get an error from L014
```
== [......................../MWE.sql] FAIL
L:   3 | P:   5 | L014 | Unquoted identifiers must be consistently capitalised.
L:   4 | P:   6 | L014 | Unquoted identifiers must be consistently capitalised.
All Finished 📜 🎉!
```

### How to reproduce

See above

### Dialect

I tested with `bigquery`, `oracle` and `tsql`, and the issue occurred in all of them. I assume it applies to all dialects.

### Version

Running in a conda env on M1 Macbook with 
```
sqlfluff, version 1.3.0
```

```
python                    3.9.7                h88f2d9e_1
```

### Configuration

No custom config used.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by j-svensmark** [comment]
I suspect the logic here might be the triggering cause of this issue https://github.com/sqlfluff/sqlfluff/blob/0feb87043c17999daf0ba3f2604abdca44d273c4/src/sqlfluff/rules/L010.py#L124

**Comment by alanmcruickshank** [comment]
Yep - 100% right on why this is happening. I think this should be a fairly easy fix to find the first character where casing matters and infer from that one 👍 . @j-svensmark - would you be up for submitting a PR for this?

**Comment by j-svensmark** [comment]
Sure, please see the PR linked above.
Btw, in case anyone needs a workaround for this, explicitly setting the capitalization policy in the `.sqlfluff` config silences this linting error
```
[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
```
